components/director/internal/domain/runtime/service.go

```diff
@@ -245,6 +250,34 @@ func (s *service) Create(ctx context.Context, in model.RuntimeInput) (string, er
 		return id, errors.Wrapf(err, "while creating multiple labels for Runtime")
 	}
 
+	// The runtime is created successfully, however there can be ASAs in the parent that should be processed.
+	tnt, err := s.tenantSvc.GetTenantByID(ctx, rtmTenant)
+	if err != nil {
+		return "", errors.Wrapf(err, "while getting tenant with id %s", rtmTenant)
+	}
+
+	if len(tnt.Parent) == 0 {
+		return id, nil
+	}
+
+	ctxWithParentTenant := tenant.SaveToContext(ctx, tnt.Parent, "")
```

Comment:
The formation is in the tenant_id and the labels produced by it should be there as the LabelDef (the formation) is there.
It will be harder if we don't have the information in place (we will need it anyways and we should derive it from the tenant - more confusing)

Maybe it can be simplified, but it will change the interface much more, currently it is not that major as there are global_subaccount_id key and value which we just convert and store, if we do this then the client won't be able to read it itself after register it (not that ASAs are with embedded tenant).

Anyways, this whole entity will be removed with the formation redesign.

Reply:
yes true, no need to invest here, probably it makes sense to add some markers in the code where it  will be removed soon due to an upcoming change